tools/cldr/cldr-to-icu/src/main/java/org/unicode/icu/tool/cldrtoicu/Cldr2Icu.java

```diff
@@ -0,0 +1,74 @@
+// © 2024 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+package org.unicode.icu.tool.cldrtoicu;
+
+import org.unicode.icu.tool.cldrtoicu.ant.CleanOutputDirectoryTask;
+import org.unicode.icu.tool.cldrtoicu.ant.ConvertIcuDataTask;
+import org.unicode.icu.tool.cldrtoicu.ant.GenerateCodeTask;
+
+public class Cldr2Icu {
+    private final Cldr2IcuCliOptions options = new Cldr2IcuCliOptions();
+
+    private void convert() {
+        ConvertIcuDataTask convert = ConvertIcuDataTask.fromXml(options.xmlConfig);
+
+        convert.setCldrDir(options.cldrDataDir);
+        convert.setOutputDir(options.outDir);
+        convert.setSpecialsDir(options.specialsDir);
+        convert.setOutputTypes(options.outputTypes);
+        convert.setIcuVersion(options.icuVersion);
+        convert.setIcuDataVersion(options.icuDataVersion);
+        convert.setCldrVersion(options.cldrVersion);
+        convert.setMinimalDraftStatus(options.minDraftStatus);
+        convert.setLocaleIdFilter(options.localeIdFilter);
+        convert.setIncludePseudoLocales(options.includePseudoLocales);
+        convert.setEmitReport(options.emitReport);
+
+        convert.init();
+        convert.execute();
+    }
+
+    private void generateCode(String action) {
+        GenerateCodeTask generateCode = new GenerateCodeTask();
+
+        generateCode.setCldrDir(options.cldrDataDir);
+        generateCode.setCOutDir(options.genCCodeDir);
+        generateCode.setJavaOutDir(options.genJavaCodeDir);
+        generateCode.setAction(action);
+
+        generateCode.init();
+        generateCode.execute();
+    }
+
+    private void outputDirectories() {
+        CleanOutputDirectoryTask clean = CleanOutputDirectoryTask.fromXml(options.xmlConfig);
+
+        clean.setRoot(options.outDir);
+        clean.setForceDelete(options.forceDelete);
+
+        clean.init();
+        clean.execute();
+    }
+
+    private void clean() {
+        outputDirectories();
+        generateCode("clean");
+    }
+
+    private void generate() {
+        convert();
+        if (!options.dontGenCode) {
+            generateCode(null);
+        }
+    }
+
+    private void run(String[] args) {
+        options.processArgs(args);
+        clean();
+        generate();
+    }
+
+    public static void main(String[] args) {
+        new Cldr2Icu().run(args);
```

Comment:
nit: This distinction between `main` and `run` seems artificial. Instead of making `Cldr2Icu` a class that you instantiate, can we not just have every method here be `static`? And then pass options as an argument to the helper methods?

There's nothing inherently stateful, nor are we creating multiple instances of `Cldr2Icu`, so it doesn't "feel" right -- it doesn't seem appropriate for what an instance is (regardless of whether that is semi-idiomatic in Java).

Reply:
It just felt bad to have a class with 8 static methods.

Options (what do you find cleaner):

1. Leave as is

2. Do all the work in main, without a `run()`
```java
public static void main(String[] args) {
    Cldr2Icu self = new Cldr2Icu();
    self.options.processArgs(args);
    self.clean();
    self.generate();
}
```

3. Close to 2, but parsing arguments in a constructor (feels like an initialization)
```java

public Cldr2Icu(String[] args) {
    options.processArgs(args);
    // or even something like
    // options = Cldr2IcuCliOptions.processArgs(args);
}

public static void main(String[] args) {
    Cldr2Icu self = new Cldr2Icu(args);
    self.clean();
    self.generate();
}
```

I can also make `options` a non-member:
```java
public static void main(String[] args) {
    Cldr2IcuCliOptions options = Cldr2IcuCliOptions.processArgs(args);
    Cldr2Icu self = new Cldr2Icu();
    self.clean(options);
    self.generate(options);
}
```

Reply:
My preference is option 2. I think nothing of 8 static methods because it's a hint of statelessness. No stateful work being done , so no chance for mutation of state. I feel bad to see 8 member methods for a stateful class when no state is truly needed. It would also entail making `options` a non-member and passing it throughout, as you suggested in the last example. Overall, that's the "cleanest".

I'm okay with option 1 if you don't think it's worth the time.

My reaction is based on the feeling that it looks very OOP Java idiomatic in a way that feels unnecessarily complicated, which is how a lot of OOP code tends to look.

Reply:
Thank you. Went with option 2.

DONE.